File: src/main/java/com/rs/java/game/area/banks/DraynorBank.java
package com.rs.java.game.area.banks;

import com.rs.java.game.WorldTile;
import com.rs.java.game.area.Area;
import com.rs.java.game.area.Shape;
import com.rs.java.game.area.shapes.Polygon;

public final class DraynorBank extends Area {

	@Override
	public Area update() {
		return this;
	}

	@Override
	public String name() {
		return "Draynor Bank";
	}

	@Override
	public Shape[] shapes() {
		return new Shape[] { new Polygon(new WorldTile[] {
				new WorldTile(3088, 3240, 0),
				new WorldTile(3088, 3246, 0),
				new WorldTile(3097, 3246, 0),
				new WorldTile(3097, 3240, 0),
				new WorldTile(3088, 3240, 0),
		}) };
	}

	@Override
	public boolean member() {
		return false;
	}

	@Override
	public Environment environment() {
		return Environment.SAFEZONE;
	}

}

File: src/main/java/com/rs/kotlin/game/world/area/areas/banks/DraynorBank.kt
package com.rs.kotlin.game.world.area.areas.banks

import com.rs.java.game.WorldTile
import com.rs.kotlin.game.world.area.Area
import com.rs.kotlin.game.world.area.shape.Polygon

class DraynorBank : Area(
    Polygon(
        arrayOf(
            WorldTile(3088, 3240, 0),
            WorldTile(3088, 3246, 0),
            WorldTile(3097, 3246, 0),
            WorldTile(3097, 3240, 0),
            WorldTile(3088, 3240, 0),
        )
    )
) {
    override fun update(): Area {
        return this
    }

    override fun name(): String {
        return "Draynor Bank"
    }

    override fun member(): Boolean {
        return false
    }

    override fun environment(): Environment {
        return Environment.SAFEZONE
    }
}
